src/dbcp/transform/acp_projects.py

```diff
@@ -479,4 +479,30 @@ def _str_strip(ser):
     }
     out.rename(columns=rename_dict, inplace=True)
     out.drop(columns=cols_to_drop, inplace=True)
-    return {"acp_projects": out}
+    return out
+
+
+def _transform_acp_changelog(raw_concat_df: pd.DataFrame) -> pd.DataFrame:
+    """Create a changelog of the ACP data over time."""
+    transformed_concat_df = raw_concat_df.groupby(
+        "report_date", group_keys=False
+    ).apply(
+        _transform_acp_projects
+    )  # TODO: hopefully this groupby apply isn't too slow
+    transformed_concat_df = transformed_concat_df.sort_values(by="report_date")
+    dedupe_cols = transformed_concat_df.columns - "report_date"
+    deduped_df = raw_concat_df.drop_duplicates(subset=dedupe_cols)
```

Comment:
haven't tested any of this, but this is roughly what I'm thinking

Reply:
I'm guessing SQL with some window functions is really all you need